src/server.cpp

```diff
@@ -2153,6 +2154,26 @@ void Server::stopSound(s32 handle)
 	m_playing_sounds.erase(i);
 }
 
+void Server::stopAllSounds()
+{
+	// Loop through references
+	for (auto it = m_playing_sounds.begin(); it != m_playing_sounds.end(); it++) {
+		ServerPlayingSound &psound = it->second;
+
+		NetworkPacket pkt(TOCLIENT_STOP_SOUND, 4);
+		pkt << it->first;
+
+		for (std::unordered_set<session_t>::const_iterator si =
+				psound.clients.begin();
+				si != psound.clients.end(); ++si) {
```

Comment:
How about `for (session_t peer_id : psound.clients) {` ? That's a bit shorter.
Same for the `m_playing_sounds` loop.